### PR TITLE
[Enhancement] Add escalation policy query setting

### LINKED ISSUE #263: [Enhancement] Filter by Escalation Policy
In addition to filtering by team and service, some users of PD Live would like to filter by escalation policy.

### PR DESCRIPTION
**Pull request by gsreynolds**
## Summary
This PR closes #263 where we have included the ability to query incidents by escalation policy.

## Screenshots
Incidents assigned to Team A escalation policy
Screenshot 2022-09-09 at 12 06 08
Incidents in Team C, assigned to Team C and Team C Manager escalation policies
Screenshot 2022-09-09 at 12 01 28
Incidents in Team C, assigned to Team C Manager escalation policy
Screenshot 2022-09-09 at 12 01 43
Filters are additive to existing selections, so filtering for Team A and Team B, but additionally selecting Team B escalation policy will only show incidents assigned to that escalation policy
Screenshot 2022-09-09 at 12 02 01
Further refining to only show incidents assigned to Team B escalation policy and Service B2
Screenshot 2022-09-09 at 12 02 58

### CODE REVIEW

File: src/redux/query_settings/sagas.js [hunk 1]
```diff
@@ -228,6 +247,7 @@ export function* validateIncidentQueryImpl() {
     if (incidentStatus) params['statuses[]'] = incidentStatus;
     if (incidentUrgency) params['urgencies[]'] = incidentUrgency;
     if (teamIds.length) params['team_ids[]'] = teamIds;
+    if (escalationPolicyIds.length) params['escalation_policy_ids[]'] = escalationPolicyIds;
```
Comment (giranm) [inline src/redux/query_settings/sagas.js:250]: I don't believe you can query by [EP ids](https://developer.pagerduty.com/api-reference/9d0b4b12e36f9-list-incidents), so this is redundant. 

We may have to update the query confirmation modal that it is an estimated number of results as we can't query this via the API (for another PR at another time!)

File: src/redux/query_settings/sagas.test.js [hunk 2]
```diff
@@ -229,4 +248,43 @@ describe('Sagas: Query Settings', () => {
         expect(result.storeState.status).toEqual(VALIDATE_INCIDENT_QUERY_REQUESTED);
       });
   });
+
+  it('updateQuerySettingsTeams', () => {
```
Comment (giranm) [inline src/redux/query_settings/sagas.test.js:252]: ❤️ this!

File: src/redux/query_settings/sagas.js [hunk 3]
```diff
@@ -158,6 +160,22 @@ export function* updateQuerySettingsTeamsImpl(action) {
...
 
+export function* updateQuerySettingsEscalationPolicies() {
+  yield takeLatest(
+    UPDATE_QUERY_SETTINGS_ESCALATION_POLICIES_REQUESTED,
+    updateQuerySettingsEscalationPoliciesImpl,
+  );
+}
+
+export function* updateQuerySettingsEscalationPoliciesImpl(action) {
+  // Update service ids and re-request incidents list + notes
```
Comment (giranm) [inline src/redux/query_settings/sagas.js:171]: Typo in comment.

File: src/redux/incidents/actions.js [hunk 4]
```diff
@@ -101,6 +105,11 @@ export const filterIncidentsByTeam = (teamIds = []) => ({
   teamIds,
 });
 
+export const filterIncidentsByEscalationPolicy = (escalationPolicyIds = []) => ({
+  type: FILTER_INCIDENTS_LIST_BY_SERVICE,
```
Comment (giranm) [inline src/redux/incidents/actions.js:109]: Needs to be `FILTER_INCIDENTS_LIST_BY_ESCALATION_POLICY`

**Review by giranm — CHANGES_REQUESTED** [review]
Great work @gsreynolds on your first enhancement.
Have made some comments/suggestions in this review.

File: src/redux/query_settings/sagas.js (hunk 1 above)
Comment (gsreynolds) [reply src/redux/query_settings/sagas.js:250]: Agree, that's redundant. I added `filterIncidentsByEscalationPolicyImpl` in `getIncidents()` similar to the Priority filter so it gets filtered after the initial GET.

```suggestion
```

File: src/redux/incidents/actions.js (hunk 4 above)
Comment (gsreynolds) [reply src/redux/incidents/actions.js:109]:
```suggestion
  type: FILTER_INCIDENTS_LIST_BY_ESCALATION_POLICY,
```

File: src/redux/query_settings/sagas.js (hunk 3 above)
Comment (gsreynolds) [reply src/redux/query_settings/sagas.js:171]:
```suggestion
  // Update escalation policy ids and re-request incidents list + notes
```

File: src/redux/query_settings/sagas.js [hunk 5]
```diff
@@ -213,6 +231,7 @@ export function* validateIncidentQueryImpl() {
       incidentStatus,
       incidentUrgency,
       teamIds,
+      escalationPolicyIds,
```
Comment (giranm) [inline src/redux/query_settings/sagas.js:234]: Need to remove this as well as this is causing ESLint to fail.

**Review by giranm — APPROVED** [review]
LGTM @gsreynolds! 🎉

**Pull request merged**